vnext/JSI.Desktop.UnitTests/JsiRuntimeUnitTests.cpp

```diff
@@ -0,0 +1,1072 @@
+
+    std::vector<PropNameID> getPropertyNames(Runtime& rt) override {
+      EXPECT_EQ(&rt, &rt_);
+      return {};
+    }
+
+    Runtime& rt_;
+  };
+
+  Object srho = Object::createFromHostObject(
+    rt, std::make_shared<SameRuntimeHostObject>(rt));
+  // Test get's Runtime is as expected
+  function("function (obj) { return obj.isSame; }").call(rt, srho);
+  // ... and set
+  function("function (obj) { obj['k'] = 'v'; }").call(rt, srho);
+  // ... and getPropertyNames
+  function("function (obj) { for (k in obj) {} }").call(rt, srho);
+
+  class TwiceHostObject : public HostObject {
+    Value get(Runtime& rt, const PropNameID& sym) override {
+      return String::createFromUtf8(rt, sym.utf8(rt) + sym.utf8(rt));
+    }
+
+    void set(Runtime&, const PropNameID&, const Value&) override {}
+  };
+
+  Object tho =
+    Object::createFromHostObject(rt, std::make_shared<TwiceHostObject>());
+  EXPECT_TRUE(function("function (obj) { return obj.abc == 'abcabc'; }")
+    .call(rt, tho)
+    .getBool());
+  EXPECT_TRUE(function("function (obj) { return obj['def'] == 'defdef'; }")
+    .call(rt, tho)
+    .getBool());
+  EXPECT_TRUE(function("function (obj) { return obj[12] === '1212'; }")
+    .call(rt, tho)
+    .getBool());
+  EXPECT_TRUE(tho.isHostObject(rt));
+  EXPECT_TRUE(
+    std::dynamic_pointer_cast<ConstantHostObject>(tho.getHostObject(rt)) ==
+    nullptr);
+  EXPECT_TRUE(tho.getHostObject<TwiceHostObject>(rt).get() != nullptr);
+
+  class PropNameIDHostObject : public HostObject {
+    Value get(Runtime& rt, const PropNameID& sym) override {
+      if (PropNameID::compare(rt, sym, PropNameID::forAscii(rt, "undef"))) {
+        return Value::undefined();
+      }
+      else {
+        return PropNameID::compare(
+          rt, sym, PropNameID::forAscii(rt, "somesymbol"));
+      }
+    }
+
+    void set(Runtime&, const PropNameID&, const Value&) override {}
+  };
+
+  Object sho = Object::createFromHostObject(
+    rt, std::make_shared<PropNameIDHostObject>());
+  EXPECT_TRUE(sho.isHostObject(rt));
+  EXPECT_TRUE(function("function (obj) { return obj.undef; }")
+    .call(rt, sho)
+    .isUndefined());
+  EXPECT_TRUE(function("function (obj) { return obj.somesymbol; }")
+    .call(rt, sho)
+    .getBool());
+  EXPECT_FALSE(function("function (obj) { return obj.notsomuch; }")
+    .call(rt, sho)
+    .getBool());
+
+  class BagHostObject : public HostObject {
+  public:
+    const std::string& getThing() {
+      return bag_["thing"];
+    }
+
+  private:
+    Value get(Runtime& rt, const PropNameID& sym) override {
+      if (sym.utf8(rt) == "thing") {
+        return String::createFromUtf8(rt, bag_[sym.utf8(rt)]);
+      }
+      return Value::undefined();
+    }
+
+    void set(Runtime& rt, const PropNameID& sym, const Value& val) override {
+      std::string key(sym.utf8(rt));
+      if (key == "thing") {
+        bag_[key] = val.toString(rt).utf8(rt);
+      }
+    }
+
+    std::unordered_map<std::string, std::string> bag_;
```

Comment:
>bag_ [](start = 49, length = 4)

(nit) why not m_? Are we all in agreement on the coding convention for Microsoft Git projects?

Reply:
I think we try to follow Facebook/react-native's conventions where we can.